### PR TITLE
Implement OpenID Connect (OIDC) Possibilities for uPortal(#2587)

### PR DESCRIPTION
**Pull request by 555vedant**
this pull request outlines OIDC implementation in uPortal. It covers OIDC endpoints, group and claims filters, and OAuth endpoint usage for clients, aiding developers in secure user authentication and profile information retrieval.

Key features:

1)Detailed descriptions and examples of OIDC endpoints.

2)Configuration for group and claims filters.

3)Client registration and scope definitions.

4)Security considerations for token management.

References:

1)Official OIDC and OAuth 2.0 specifications.

2)uPortal documentation links.

### CODE REVIEW

**Comment by 555vedant** [comment]
@jonathanmtran please suggest me necessary changes!

File: docs/OIDC/README.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+- **Revocation**: Implement token revocation to invalidate tokens when necessary.
+- **Secure Storage**: Store client secrets and tokens securely to prevent unauthorized access.
+
+## References
+- [OIDC Official Documentation](https://openid.net/developers/specs/)
+- [OAuth 2.0 Specification](https://oauth.net/2/)
+- [uPortal Documentation](https://apereo.github.io/uPortal/)
+
+## Conclusion
+This documentation serves as a comprehensive guide for developers to understand and implement OIDC using OAuth 2.0 in uPortal, facilitating secure user authentication and resource access.
```
Comment (ChristianMurphy) [inline docs/OIDC/README.md:135]: @555vedant this feels very ChatGPT like... 🤔

File: docs/OIDC/README.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+### Scopes and Permissions
+- **Definition**: Scopes determine the level of access requested by the client.
+- **Example**: 
+    ```json
+    "scope": "openid profile email"
+    ```
+- **Common Scopes**:
+    - `openid`: Required to perform OIDC authentication.
+    - `profile`: Requests access to the user's profile information.
+    - `email`: Requests access to the user's email address.
```
Comment (ChristianMurphy) [inline docs/OIDC/README.md:103]: uPortal has custom claims, this looks like generic OIDC

File: docs/OIDC/README.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+### Group Filters
+- **Description**: Specify which user groups are included in the OIDC token.
+- **Example**: 
+    ```json
+    "groups": ["admin", "user"]
+    ```
+
+### Claims Filters
+- **Description**: Define which claims are returned in the ID token or access token.
+- **Example**:
```
Comment (ChristianMurphy) [inline docs/OIDC/README.md:77]: it looks like the markdown formatting got messed up here

File: docs/OIDC/README.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
+# OpenID Connect (OIDC) Implementation for uPortal
+
+## Introduction
+OpenID Connect (OIDC) is an authentication layer built on top of OAuth 2.0, enabling clients to verify user identities and obtain basic profile information. This documentation outlines the OIDC possibilities, including available endpoints, group and claims filters, and using OAuth endpoints for clients.
```
Comment (ChristianMurphy) [inline docs/OIDC/README.md:4]: I don't think this needs an introduction, we can link out to the official OIDC docs or wikipedia for this.
Also uPortal specifically uses form of three legged auth, which doesn't seem to be referenced anywhere

**Review by ChristianMurphy — CHANGES_REQUESTED** [review]
Thanks for starting on this @555vedant!
While it is okay to use ChatGPT to help generate content, it is good to check that it is accurate/correct yourself, and not ask maintainers to do all the manual fact checking, grammar checking, and readability checking.

**Comment by 555vedant** [comment]
@ChristianMurphy sir  i am closing this PR i dont want to make noise in this file i will generate another PR

**Comment by 555vedant** [comment]
@ChristianMurphy i am closing this PR i dont wanted to make this file noise  i will make it from scratch !!

**Pull request closed without merging**